grails-gradle-plugin/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy

```diff
@@ -407,4 +413,63 @@ class GrailsGradlePlugin extends GroovyPlugin {
         native2asciiTask
     }
 
+    protected Jar createPathingJarTask(Project project, String name, Configuration...configurations) {
+        File assets = new File(project.buildDir, 'assets')
+
+        project.tasks.create(name, Jar) { Jar task ->
+            task.dependsOn(configurations)
+            task.appendix = 'pathing'
+
+            Set files = []
+            configurations.each {
+                files.addAll(it.files)
+            }
+
+            task.doFirst {
+                manifest { Manifest manifest ->
+                    manifest.attributes "Class-Path": files.collect { File file ->
+                        file.toURI().toURL().toString().replaceFirst(/file:\/+/, '/')
+                    }.join(' ')
+                }
+            }
+
+            // assetCompile will be execute for all Jar-type tasks
+            // (see https://github.com/bertramdev/asset-pipeline/blob/master/asset-pipeline-gradle/src/main/groovy/asset/pipeline/gradle/AssetPipelinePlugin.groovy#L85)
+            // exclude the assets from pathing jar
+            task.exclude {
+                it.file.absolutePath.startsWith(assets.absolutePath)
+            }
+        }
+    }
+
+    protected void configurePathingJar(Project project) {
+        project.afterEvaluate {
+            ConfigurationContainer configurations =  project.configurations
+            Configuration runtime = configurations.getByName('runtime')
+            Configuration console = configurations.getByName('console')
+
+            Jar pathingJar = createPathingJarTask(project, "pathingJar", runtime)
+            Jar pathingJarCommand = createPathingJarTask(project, "pathingJarCommand", runtime, console)
+
+            GrailsExtension grailsExt = project.extensions.getByType(GrailsExtension)
+
+            if (grailsExt.pathingJar && Os.isFamily(Os.FAMILY_WINDOWS)) {
```

Comment:
Ya I originally had it that way. I wasn't sure if there was a possible use case where someone would want to run it standalone or have a custom task that depends on it...